### ISSUE TITLE
Drt waiting times in drt-pt scenario

### ISSUE THREAD

**Issue by helgehr** [matsim-org/matsim-code-examples#457]
Hello, I have a problem with the waiting times in a scenario where people get transport via intermodal access/egress to a public transport system.
When I run a simulation without public transport (1e5 requests) I get the following reasonable drt waiting times:
![0 waitStats_drt](https://user-images.githubusercontent.com/38046421/98034114-c64d4a80-1e16-11eb-9ec1-8af0086d5d7d.png)

If I do the same with a scenario combining drt and pt (same population) I get the following drt waiting times:

![0 waitStats_drt](https://user-images.githubusercontent.com/38046421/98034456-512e4500-1e17-11eb-912e-6957a4b2b9fd.png)

Does anyone of you drt experts (maybe @michalmac @sebhoerl @jfbischoff) have an explanation why the waiting time in the combined scenario builds up over a long time and why it is on a much higher level in general?

**Comment by helgehr** [comment]
I noticed when the same drt access/egress trips are computed beforehand and the swiss rail raptor parameter `useIntermodalAccessEgress` is set to false the following picture for the waiting times is observed:

![0 waitStats_drt](https://user-images.githubusercontent.com/38046421/98094878-29c49000-1e8a-11eb-90a0-7cccae8e271e.png)

So it must have something to do with the swiss rail raptor Intermodal AccessEgress Module Component, do you know what could be a reason for this behaviour?

**Comment by michalmac** [comment]
Ping @vsp-gleich

**Comment by vsp-gleich** [comment]
Maybe pure drt trips in addition to drt access/egress to pt trips are too many for the limited fleet, so a higher wait time builds up? The intermodal router part only calls the drt router and takes over what the drt router returns. In mobsim drt access/egress to pt and pure drt trips should be treated equal. Both are normal drt legs, there is nothing special about drt access/egress vs. pure drt trips in the plans. Consequently they should appear equal to the drt insertion algorithm etc. in the mobsim. So the most obvious reason for higher wait is higher demand due to making it feasible for more agents to take drt.

**Comment by jfbischoff** [comment]
I would agree with @vsp-gleich: How does the overall vehicle occupancy change when you enable intermodal access / egress?
One other reason could be the distribution of vehicles, which could be improved by enabling re-balancing or modifying the initial start links of the simulation.

**Comment by helgehr** [comment]
So in the mentioned scenario are 1000 drt vehicles and the average occupancy is low (see attached picture). I also tried to rebalancing ("minCostFlowRebalancingStrategy") but it did not make a difference.
![0 drt_occupancy_time_profiles_StackedArea_drt_small](https://user-images.githubusercontent.com/38046421/99398029-1da5ed00-28e4-11eb-8622-5044a1f0aa59.png)
I suspect that somehow the search for a close station in the intermodal access/egress module is computationally heavy and so it takes some time to look for stations ad hoc. Therefore the wait time builds up until a certain steady state for incoming access/egress trips is reached.
So the problem presumably lies in the swiss rail raptor module and not the drt contribution.

**Comment by vsp-gleich** [comment]
There is no ad hoc station search. Intermodal routing and drt routing all happen during the replanning stage, i.e. during mobsim there is only the assignment of which drt vehicle shall serve the ride request, but the agent already has fixed departure and arrival drt links or stops. Simulation speed is not necessarily stable, the simulation would wait for any part still calculating.
Please keep in mind that drt vehicle occupancy can change over the course of iterations. Demand can be high at the beginning, then agents learn drt is congested and makes for long waits and after innovation switch off there is an uncongested drt fleet.

**Comment by helgehr** [comment]
Ok, that is good to know - thank you!
The pictures and my simulation is only for one iteration/day. Innovation only happens from iteration to iteration and not within an iteration, right?
